### DISCUSSION TITLE
How to preserve the working directory when running a pnpm command

### DISCUSSION THREAD

**Discussion by janniksam** [pnpm/pnpm#7670] [Q&A]
For example this is my structure:
/project/package.json
/project/my/deep/dir

I am calling `pnpm mycommand` inside "/project/my/deep/dir."
The command `mycommand` always gets  /project as the working dir.

Is there an easy way to preserve "/project/my/deep/dir" as a working dir in mycommand?

**Comment by matoruru** [comment]
@janniksam Unfortunately there's no official solution for this.

Here is the workaround:

In `package.json`
```
"scripts": {
  "mycommand": "cd ${WORKDIR} && pwd"
},
```

Execute the pnpm command like this:

```
WORKDIR=$(pwd) pnpm run mycommand
```

This will let you get the nested directory as the result...

**Reply by b3nab** [reply]
How is it possible? I was looking for a similar solution and found out this: https://pnpm.io/pnpm-cli stating that options -C or --dir  `Run as if pnpm was started in <path> instead of the current working directory.` .

But it's not working and for the moment and the only solution is your workaround.
